File: Algorithms/src/3550. Smallest Index With Digit Sum Equal to Index/Code/solution.go
func smallestIndex(nums []int) int {
    for i := 0; i < len(nums); i++ {
        x := nums[i]
        sum := 0
        for x > 0 {
            sum += x % 10
            x /= 10
        }
        if sum == i {
            return i
        }
    }
    return -1
}

File: Algorithms/src/3550. Smallest Index With Digit Sum Equal to Index/Code/solution.java
class Solution {
    public int smallestIndex(int[] nums) {
        for (int i = 0; i < nums.length; ++i) {
            int x = nums[i];
            int sum = 0;
            while (x > 0) {
                sum += x % 10;
                x /= 10;
            }
            if (sum == i) return i;
        }
        return -1;
    }
}